src/helpers/object_tagger_helper.py

```diff
@@ -0,0 +1,100 @@
+from helpers import aws_helper, console_printer
+from typing import List, Dict
+
+
+def aws_tags_are_subset(
+    subset: List[Dict],
+    superset: List[Dict],
+):
+    subset_tags_flat = {tag["Key"]: tag["Value"] for tag in subset}
+    superset_tags_flat = {tag["Key"]: tag["Value"] for tag in superset}
+
+    for subset_key, subset_value in subset_tags_flat.items():
+        if subset_key not in superset_tags_flat:
+            return False
+        if superset_tags_flat[subset_key] != subset_value:
+            return False
+
+    return True
+
+
+def verify_s3_object_required_tags(s3_bucket, s3_object_key, tags_dict):
+    # infer table/db from s3 object key
+    console_printer.print_info(f"KEY: {s3_object_key}")
+    split_string = s3_object_key.split("/")
+    if len(split_string) < 3:
+        console_printer.print_warning_text(
+            f"Key doesn't match pattern: {s3_object_key}"
+        )
+        return
+
+    # cater for keys ending "_$folder$"
+    if split_string[-1].endswith("_$folder$"):
+        split_string[-1] = split_string[-1][0:-9]
+
+    # replace any strings ending `.db`
+    if any(".db" in item for item in split_string):
+        for index, value in enumerate(split_string):
+            split_string[index] = (
+                value.replace(".db", "") if value.endswith(".db") else value
+            )
+
```

Comment:
From here, can we split up the logic below into other functions? Im thinking about the standalone object_tagger tests at this stage and also future tests that might require similar functionality.

That said, would it make sense to have a function, encapsulating what you have below, where you can pass in a db_name, table_name & the tags_dict to populate the **required_tags_list**, the aws_object_tags_list and return it back to the test. At that point make the assertion within the test code itself as opposed to making the assertion in a helper (for improved readability)?